components/fetch-build-scan-data-cmdline-tool/src/main/java/com/gradle/enterprise/model/BuildTimeMetrics.java

```diff
@@ -0,0 +1,47 @@
+package com.gradle.enterprise.model;
+
+import java.time.Duration;
+
+public class BuildTimeMetrics {
+
+    public final Duration initialBuildTime;
+    public final Duration instantSavings;
+    public final Duration instantSavingsBuildTime;
+    public final Duration pendingSavings;
+    public final Duration pendingSavingsBuildTime;
+
+    private BuildTimeMetrics(
+            Duration initialBuildTime,
+            Duration instantSavings,
+            Duration instantSavingsBuildTime,
+            Duration pendingSavings,
+            Duration pendingSavingsBuildTime) {
+        this.initialBuildTime = initialBuildTime;
+        this.instantSavings = instantSavings;
+        this.instantSavingsBuildTime = instantSavingsBuildTime;
+        this.pendingSavings = pendingSavings;
+        this.pendingSavingsBuildTime = pendingSavingsBuildTime;
+    }
+
+    public static BuildTimeMetrics from(BuildScanData firstBuild, BuildScanData secondBuild) {
+        if (firstBuild.getBuildTime() == null ||
+                secondBuild.getBuildTime() == null ||
+                secondBuild.getExecutedCacheableSummary() == null ||
+                secondBuild.getSerializationFactor() == null) {
+            return null;
```

Comment:
It might be better to return `Optional<BuildTimeMetrics>` rather than using `null`.